_posts/2025-09-30-add-new-table-to-capture-list.adoc

```diff
@@ -0,0 +1,143 @@
+This is why Debezium requires capturing the database schema at first start-up.
+
+As a side note, the initial snapshot exists primarily to capture the current data present in the database,
+since not all data may be available in the transaction log when Debezium starts for the very first time.
+Capturing the schema is required so that Debezium knows the table structures.
+Even if you don’t want to capture the existing data, schema capture cannot be disabled.
+You can, however, skip capturing the data itself by using the no_data snapshot mode.
+
+That said, you should start to understand why the answer is “it depends.”
+
+Let’s walk through a couple of common scenarios you may run into.
+
+=== Scenario 1
+
+Let’s start with an easy scenario: you’ve set the `*.include.list` properties and ended up capturing all available tables in the database.
+
+First of all, in this scenario, the `schema.history.internal.store.only.captured.tables.ddl` property has no impact, since all table schemas are captured anyway.
+
+If you create a new table while Debezium is streaming, it captures the DDL and updates its internal schema representation.
+When DML occurs on that table, Debezium starts emitting change events immediately, since the table structure is already known.
+
+NOTE: You need to check that the database configuration for CDC is set up so that a new table is automatically enabled for CDC.
+This depends on the connector specifics.
+For example, in PostgreSQL, this means that `publication.autocreate.mode` is either `all_tables` (the default) or disabled, (and you manually created it for all tables).
+
+=== Scenario 2
+
+Suppose now that we are in a scenario where the `*.include.list` parameters are set so that Debezium captures only a subset of the tables available in the database.
+If you leave `schema.history.internal.store.only.captured.tables.ddl` untouched (`false` by default), Debezium will retrieve all table schemas during the initial snapshot.
+
+So, if you want to start capturing changes from a table that wasn’t previously captured, you need to:
+
+. Edit the `table.include.list` property to specify the tables you want to capture.
+. Restart the connector.
+. [Optional] Initiate an `incremental snapshot` if you want to capture existing data from the newly added tables.
+
+The process is slightly more complex if you set `schema.history.internal.store.only.captured.tables.ddl` to `true`.
+In that case, since Debezium has no information about the table you want to add, you need to explicitly tell the connector to retrieve it.
+
+The procedure is:
+
+. Stop the connector.
+. Remove the internal database schema history topic specified by the `schema.history.internal.kafka.topic` property.
+. Set the `snapshot.mode` property to `recovery` (or `no_data` for connectors that don’t support recovery, like PostgreSQL).
```

Comment:
It might be good to highlight the "no schema changes" prerequisite here when using `recovery`, just to avoid any potential issues with people who use the blog as the truth rather than consult the docs.